evalml/tests/pipeline_tests/test_component_graph.py

```diff
@@ -1252,3 +1252,35 @@ def test_component_graph_sampler_same_given_components(mock_dt_fit, mock_rf_fit)
     pd.testing.assert_series_equal(
         mock_dt_fit.call_args[0][1], mock_rf_fit.call_args[0][1]
     )
+
+
```

Comment:
@chukarsten Yeah honestly I'm not sure there's a good component graph example with the same component dictionary but different compute order, since we generate the order using the component dict. I guess one way could be artifically made via manually changing the compute_order attribute after its been set?

Reply:
I think this is an example:

```python
cg = ComponentGraph({
    "Imputer": [Imputer],
    "OneHot": [OneHotEncoder, "Imputer.x"],
    "DT": [DateTimeFeaturizer, "Imputer.x"],
    "Logistic Regression": [
        LogisticRegressionClassifier,
        "OneHot.x",
        "DT.x",
    ],
})

cg2 = ComponentGraph({
    "Imputer": [Imputer],
    "DT": [DateTimeFeaturizer, "Imputer.x"],
    "OneHot": [OneHotEncoder, "Imputer.x"],
    "Logistic Regression": [
        LogisticRegressionClassifier,
        "OneHot.x",
        "DT.x",
    ],
})

assert cg.component_dict == cg2.component_dict
assert cg.compute_order != cg2.compute_order
assert cg.random_seed == cg2.random_seed
assert cg != cg2
```

@angela97lin Can you add coverage for this case? Is this the intended behavior though? The two graphs should be equivalent right? 

Might also be good to add coverage for this case, which the graphs are equal in this case:

```python
cg = ComponentGraph({
        "Component A": [DateTimeFeaturizer],
        "Component B": [OneHotEncoder],
        "Random Forest": [RandomForestClassifier, "Component A.x", "Component B.x"],
    })

cg2 = ComponentGraph({
        "Component B": [OneHotEncoder],
        "Component A": [DateTimeFeaturizer],
        "Random Forest": [RandomForestClassifier, "Component A.x", "Component B.x"],
    })

cg2 == cg
```

Reply:
@freddyaboulton Hmm, good call. I guess in this case, we should make the decision here: do we want to consider these two component graphs equivalent? Functionally, they should be... though I guess we do calculate things in a different order. If we decide they are equivalent, then maybe its best to remove the check for compute order. If we decide they're not equivalent, then this would be a great test to add.

Reply:
I think users primarily care about whether the edges and nodes are equal. The compute order seems like an implementation detail to me. But open to hearing arguments for the other option!

Reply:
I still think we should add both repros to our tests no matter what we decide though?

Reply:
@freddyaboulton 
I agree--though with the way we currently have things structured, we have `compute_order` as something that users can change, and that feels weird. I think we should:
1. Keep logic as is (compute order must be equivalent) for graphs to be equivalent. I'll add the test above, thank you for that :)
2. File an issue to make compute_order read-only / not changeable, since we can end up in a weird state if users decide to change the attribute but not the edges/nodes in the graph. Does that sound good?

Here's the filed issue: https://github.com/alteryx/evalml/issues/2342

Reply:
Sounds good to me! Thanks @angela97lin